### PR TITLE
Support non-fatal errors in status check

### PR DESCRIPTION
**Pull request by lukasmacko**

### CODE REVIEW

File: health/probe/plugin_impl_probe.go [hunk 1]
```diff
@@ -101,8 +113,8 @@ func (p *Plugin) readinessProbeHandler(formatter *render.Render) http.HandlerFun
 // livenessProbeHandler handles k8s liveness probe.
 func (p *Plugin) livenessProbeHandler(formatter *render.Render) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
-		stat := p.StatusCheck.GetAgentStatus()
-		statJSON, _ := json.Marshal(p.StatusCheck.GetAgentStatus())
+		stat := p.getAgentStatus()
+		statJSON, _ := json.Marshal(stat)
```
Comment (VladoLavor) [inline health/probe/plugin_impl_probe.go]: any reason to ignore error here?

File: health/probe/plugin_impl_probe.go [hunk 2]
```diff
@@ -113,3 +125,41 @@ func (p *Plugin) livenessProbeHandler(formatter *render.Render) http.HandlerFunc
...
+				}
+			}
+
+		}
+	}
+
+	return exposedStatus
+}
+
+func isInSlice(haystack []string, needle string) bool {
```
Comment (VladoLavor) [inline health/probe/plugin_impl_probe.go]: I would rather use some context-related naming

File: health/probe/plugin_impl_probe.go (hunk 1 above)
Comment (lukasmacko) [reply health/probe/plugin_impl_probe.go]: It was ignored before and I'm not sure whether it's worth to log it or do something about it. Any suggestion?

File: health/probe/plugin_impl_probe.go (hunk 2 above)
Comment (lukasmacko) [reply health/probe/plugin_impl_probe.go]: similar function https://golang.org/pkg/sort/#SearchInts uses single letter names, should I rename it?

**Pull request merged**